### PR TITLE
folly: fix hooks

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **folly/***

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/folly/all/conanfile.py [hunk 1]
```diff
@@ -76,10 +60,10 @@ def requirements(self):
         self.requires("glog/0.4.0")
         self.requires("libevent/2.1.12")
         self.requires("lz4/1.9.3")
-        self.requires("openssl/1.1.1i")
+        self.requires("openssl/1.1.1k")
```
Comment (madebr) [inline recipes/folly/all/conanfile.py:63]:
```suggestion
        self.requires("openssl/1.1.1l")
```

**Review by madebr — APPROVED** [review]

File: recipes/folly/all/conanfile.py [hunk 2]
```diff
@@ -47,26 +47,10 @@ def configure(self):
...
-        if self.settings.os in ["Macos", "Windows"] and self.options.shared:
-            raise ConanInvalidConfiguration("Folly could not be built on {} as shared library".format(self.settings.os))
-
-        if self.version == "2020.08.10.00" and self.settings.compiler == "clang" and self.options.shared:
-            raise ConanInvalidConfiguration("Folly could not be built by clang as a shared library")
 
     # Freeze max. CMake version at 3.16.2 to fix the Linux build
     def build_requirements(self):
-        self.build_requires("cmake/3.16.2")
+        self.build_requires("cmake/3.16.9")
```
Comment (ericriff) [inline recipes/folly/all/conanfile.py:53]: There is a comment just above this that says
```
    # Freeze max. CMake version at 3.16.2 to fix the Linux build
```
:thinking:

File: recipes/folly/all/test_package/test_package.cpp [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
-    folly::Future<std::string> future = promise.getSemiFuture().via(&executor);
-    folly::Future<folly::Unit> unit = std::move(future).thenValue(print_uri);
-    promise.setValue("https://github.com/bincrafters");
-    std::move(unit).get();
-#if FOLLY_HAVE_ELF
-    folly::symbolizer::ElfFile elffile;
-#endif
-    return EXIT_SUCCESS;
-}
+#include <cstdlib>
```
Comment (ericriff) [inline recipes/folly/all/test_package/test_package.cpp:1]: What did you change here?
Github is rendering the diff weirdly.

**Review by ericriff — APPROVED** [review]

File: recipes/folly/all/test_package/test_package.cpp (hunk 3 above)
Comment (ericLemanissier) [reply recipes/folly/all/test_package/test_package.cpp:1]: Line endings: crlf->lf

File: recipes/folly/all/conanfile.py (hunk 2 above)
Comment (ericLemanissier) [reply recipes/folly/all/conanfile.py:53]: I guess it's OK as long as we don't cross 3.17.0

Comment (ericriff) [reply recipes/folly/all/conanfile.py:53]: Should we update the comment then?

**Review by ericriff — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**